src/main/java/bob/Bob.java

```diff
@@ -0,0 +1,237 @@
+            switch (s.substring(0,3)) {
+            case "[T]":
+                list.add(new Task(s));
+                break;
+            case "[D]":
+                list.add(new Deadline(s));
+                break;
+            case "[E]":
+                list.add(new Event(s));
+                break;
+            }
+        }
+
+        return list;
+    }
+
+    private static boolean processCommand(String command, String line, List<Task> list, String filename) throws BobException {
+        switch (command) {
+        case "todo":
+            addTodo(line, list);
+            break;
+        case "deadline":
+            addDeadline(line, list);
+            break;
+        case "event":
+            addEvent(line, list);
+            break;
+        case "mark":
+            markTask(line, list);
+            break;
+        case "unmark":
+            unmarkTask(line, list);
+            break;
+        case "list":
+            displayList(list);
+            break;
+        case "save":
+            saveList(filename, list);
+            break;
+        case "bye":
+            displayExitMessage();
+            return true;
+        default:
+            throw new BobException("I'm sorry, but I don't know what that means :-(");
+        }
+        return false;
+    }
+
+    private static void saveList(String filename, List<Task> list) throws BobException {
+
+        // convert list of tasks objects to list of strings
+        List<String> stringList = list.stream()
+                .map(object -> Objects.toString(object, null))
+                .collect(Collectors.toList());
+
+        Path file = Paths.get(filename);
+
+        try {
+            Files.write(file, stringList, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new BobException("Some problem with your computer, could not save list.");
+        }
+
+        displayHorizontalLine();
+        System.out.println("Your list has been saved to " + filename + ".");
+        displayHorizontalLine();
+    }
+
+    private static void unmarkTask(String line, List<Task> list) {
+        String content;
+        content = line.split(" ", 2)[1];
+        list.get(Integer.parseInt(content) - 1).setDone(false);
+        displayHorizontalLine();
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println(list.get(Integer.parseInt(content) - 1).getListItem());
+        displayHorizontalLine();
+    }
+
+    private static void markTask(String line, List<Task> list) {
+        String content;
+        content = line.split(" ", 2)[1];
+        list.get(Integer.parseInt(content) - 1).setDone(true);
+        displayHorizontalLine();
+        System.out.println("Nice! I've marked this task as done: ");
+        System.out.println(list.get(Integer.parseInt(content) - 1).getListItem());
+        displayHorizontalLine();
+    }
+
+    private static void addEvent(String line, List<Task> list) throws BobException {
```

Comment:
try to have a single level of abstraction by abstracting out chunks of code into different functions

apply this to the othe functions in ths class too, you might find that there are some redundancies and it would be easy to extract these out into separate functions. 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard